aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncClient.java

```diff
@@ -173,22 +197,47 @@ public static AppSyncClient via(@NonNull GraphQLBehavior api) {
         return new NoOpCancelable();
     }
 
+    /**
+     * Uses Amplify API to make a mutation which will only apply if the version sent matches the server version.
+     *
+     * @param clazz      The class of the object being deleted
+     * @param objectId   ID id of the object to delete
+     * @param version    The version of the model we have
+     * @param onResponse Invoked when response data is available.
+     * @param onFailure  Invoked on failure to obtain response data
+     * @return A {@link Cancelable} to provide a means to cancel the asynchronous operation
+     */
+    @NonNull
+    @Override
+    public <T extends Model> Cancelable delete(
+            @NonNull Class<T> clazz,
+            @NonNull String objectId,
+            @NonNull Integer version,
+            @NonNull Consumer<GraphQLResponse<ModelWithMetadata<T>>> onResponse,
+            @NonNull Consumer<DataStoreException> onFailure) {
+        return delete(clazz, objectId, version, null, onResponse, onFailure);
+    }
+
     @NonNull
     @Override
     public <T extends Model> Cancelable delete(
             @NonNull Class<T> clazz,
             @NonNull String objectId,
             @NonNull Integer version,
+            @Nullable QueryPredicate predicate,
             @NonNull Consumer<GraphQLResponse<ModelWithMetadata<T>>> onResponse,
             @NonNull Consumer<DataStoreException> onFailure) {
         try {
-            final String doc = AppSyncRequestFactory.buildDeletionDoc(clazz);
+            final String doc = AppSyncRequestFactory.buildDeletionDoc(clazz, predicate != null);
```

Comment:
Can you extract a local here to communicate what this bool does? Alternatively, should we pass something other than a bool here to avoid the construction here on 238?

Reply:
Daniel and I agree we should just pass down the predicate, here. Even if it just means `predicate != null` is checked as a method-local implementation detail.

Alternately, I am always tryign to get people to do:
```
enum LabeledBoolean {
    STATE_A,
    STATE_B
}
```
with 
```
exerciseBehavior(LabeledBoolean.STATE_A);
```
It's a more legible experience to read that, than `exerciseBeahvior(false)` -- the `false` is meaningless until I read the code.

Reply:
I had something like that stashed away somewhere but can't seem to find it. I'll get it changed though.